### PR TITLE
refactor: a few 'metadata' and the 'annotation' components to Typescript

### PR DESCRIPTION
**Pull request by orlinmalkja**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "standard-version": "^9.5.0",
     "start-server-and-test": "^2.0.3",
     "tailwindcss": "^3.4.1",
+    "typescript": "^5.4.5",
```
Comment (paulpestov) [inline package.json:53]: Why is this necessary? It wasn't before

File: src/components/annotations/AnnotationsList.vue [hunk 2]
```diff
@@ -21,40 +21,40 @@
...
+  types: any[]
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  activeAnnotation: () => <ActiveAnnotation>{},
+  configuredAnnotations: () => <Annotation[]> [],
+  toggle: () => null,
+})
+
+const annotationTypesMapping = computed<Object>(() => (
```
Comment (paulpestov) [inline src/components/annotations/AnnotationsList.vue:42]: you could specify a local interface here

File: package.json (hunk 1 above)
Comment (orlinmalkja) [reply package.json:53]: I removed it

File: src/components/annotations/AnnotationsList.vue (hunk 2 above)
Comment (orlinmalkja) [reply src/components/annotations/AnnotationsList.vue:42]: Here is the link to the `AnnotationsAPI` https://subugoe.pages.gwdg.de/emo/text-api/page/annotation_specs/#target-object

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,112 @@
+{
```
Comment (paulpestov) [inline tsconfig.json:1]: Is this also needed?

Comment (orlinmalkja) [reply tsconfig.json:1]: Yes I think it should be there. Its the Typescript config file

File: src/components/annotations/AnnotationsList.vue [hunk 4]
```diff
@@ -21,40 +21,40 @@
...
     return prev;
   }, {})
 ));
 
-function isActive(annotation) {
+function isActive(annotation: Annotation): boolean {
   return !!props.activeAnnotation[annotation.id];
 }
-function isText(annotation) {
+function isText(annotation: Annotation) : boolean {
```
Comment (paulpestov) [inline src/components/annotations/AnnotationsList.vue:53]: space

File: src/components/annotations/AnnotationsList.vue [hunk 5]
```diff
@@ -21,40 +21,40 @@
...
   return annotationTypesMapping.value[annotation.body['x-content-type']] === 'text';
 }
-function getIconName(typeName) {
+function getIconName(typeName: string): string {
+  // Question: is there any range of values for the typeName ? I set it to 'string' by considering examples when just printing it
```
Comment (paulpestov) [inline src/components/annotations/AnnotationsList.vue:57]: No I think there is no range since it can be configured by the users. Could you pls remov ethis comment from code?

File: src/components/annotations/AnnotationsList.vue (hunk 2 above)
Comment (orlinmalkja) [reply src/components/annotations/AnnotationsList.vue:42]: Yes, this sounds good

File: tsconfig.json (hunk 3 above)
Comment (orlinmalkja) [reply tsconfig.json:1]: I got the explaination from https://www.digitalocean.com/community/tutorials/typescript-new-project# 
Using the --init flag in the above command will initialize your project by creating a tsconfig.json file in your typescript-project project directory. This tsconfig.json file will allow you to configure further and customize how TypeScript and the tsc compiler interact. You can remove, add, and change configurations in this file to best meet your needs.

File: src/components/annotations/AnnotationsList.vue (hunk 2 above)
Comment (orlinmalkja) [reply src/components/annotations/AnnotationsList.vue]: added the interface `AnnotationTypesMapping`

File: src/components/annotations/AnnotationsList.vue (hunk 5 above)
Comment (orlinmalkja) [reply src/components/annotations/AnnotationsList.vue]: Removed the comment

File: tsconfig.json (hunk 3 above)
Comment (paulpestov) [reply tsconfig.json:1]: Ok, but we are not using the typescript compiler for now, right? Maybe we should add it later.

Comment (orlinmalkja) [reply tsconfig.json]: okay, deleted it

**Review by paulpestov — APPROVED** [review]
